snapcraft/storeapi/__init__.py

```diff
@@ -287,17 +287,36 @@ def _download_snap(self, name, channel, arch, download_path,
             return
         logger.info('Downloading {}'.format(name, download_path))
 
+        resume_possible = False
+        total_read = 0
+        probe_url = requests.head(download_url)
+        if probe_url.is_redirect:
+            download_url = probe_url.headers['Location']
+            resume_possible = True
```

Comment:
Can you explain this? Why is a resume possible only if the download URL is a redirect?